### PR TITLE
fix: Message formatting breaks telegram > discord #340 | feat: new option "useEmbeds"

### LINKED ISSUE #340: [Bug] Message formatting breaks telegram > discord
The pull request #336 introduces a new critical bug (telegram > discord).

Message formatting cuts half of the content.

CC @Exzender (could you look into this)

image

### PR DESCRIPTION
**Pull request by Exzender**
1. Full fix for #340 (When some entity in Telegram has double(triple...) formatting - parser cuts off part of source message)
2. Fix: Message edits in telegram thread were not transferred
3. Added bridge option "useEmbeds" to define embed usage (always / never / auto)

### CODE REVIEW

**Comment by Exzender** [comment]
Added bridge option "useEmbeds" to disable/enable embed usage.

**Comment by Lulalaby** [comment]
Well, amazing
image

**Comment by Exzender** [comment]
> Well, amazing image

fixed.
and added link parsing (from Discord) (#262)

**Comment by Exzender** [comment]
Сompletely rewrote the code of handleEntities to support nested entities (#340)

**Comment by Lulalaby** [comment]
btw it seems your local git config is incorrect (either user or email), you might wanna fix it

File: README.md [hunk 1]
```diff
@@ -102,6 +102,7 @@ As mentioned in the step-by-step installation guide, there is a settings file. H
 	* `discord.relayLeaveMessages`: Whether to relay messages to Telegram about people leaving the Discord chat
 	* `discord.sendUsernames`: Whether to send the sender's name with the messages to Telegram
 	* `discord.crossDeleteOnTelegram`: Whether to also delete the corresponding message on Telegram when one is deleted in Discord
+    * `discord.useEmbeds`: Whether to use embeds for current bridge. Can be `always`, `never`, `auto`. Defaults to `false`
 	* `threadMap`: An array containing all threads mapping for each bridge
```
Comment (Lulalaby) [inline README.md]:
```suggestion
	* `discord.crossDeleteOnTelegram`: Whether to also delete the corresponding message on Telegram when one is deleted in Discord
        * `discord.useEmbeds`: Whether to use embeds for current bridge. Can be `always`, `never`, `auto`. Defaults to `false`
	* `threadMap`: An array containing all threads mapping for each bridge
```

File: src/telegram2discord/setup.ts [hunk 2]
```diff
@@ -150,6 +148,7 @@ export function setup(
 
 			// Don't crash on errors
 			tgBot.catch((err: any) => {
+				console.log(err);
```
Comment (Lulalaby) [inline src/telegram2discord/setup.ts]:
```suggestion
```

File: src/telegram2discord/setup.ts [hunk 3]
```diff
@@ -130,8 +130,6 @@ export function setup(
 			tgBot.use(middlewares.addBridgesToContext);
 			tgBot.use(middlewares.informThisIsPrivateBot);
 			tgBot.use(middlewares.removeD2TBridges);
-			//@ts-ignore telegram expacts a second parameter
-			tgBot.command(middlewares.removeBridgesIgnoringCommands);
 			tgBot.on("new_chat_members", middlewares.removeBridgesIgnoringJoinMessages);
```
Comment (Lulalaby) [inline src/telegram2discord/setup.ts:135]:
```suggestion
			//@ts-ignore telegram expacts a second parameter
			//tgBot.command(middlewares.removeBridgesIgnoringCommands);
			tgBot.on("new_chat_members", middlewares.removeBridgesIgnoringJoinMessages);
```

**Review by Lulalaby — APPROVED** [review]

**Pull request merged**